zgw_consumers/migrations/0022_service_slug.py

```diff
@@ -0,0 +1,59 @@
+# Generated by Django 5.1 on 2024-08-12 13:20
+
+from django.db import migrations, models, transaction
+from django.utils.text import slugify
+
+
+def set_service_slug_default_from_api_root(apps, schema_editor):
+    Service = apps.get_model("zgw_consumers", "Service")
+
+    with transaction.atomic():
+        # Lock the table to avoid inserting any additional rows.
```

Comment:
hrm, no locking is happening here - and if you would use `select_for_update`, it would only lock the existing rows, not lock the table for new inserts.

If you're not doing any locking, I think this transaction block is excessive since migrations already run in a transaction on databases that support them.